pkg/virt-handler/virtwrap/isolation/isolation.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package isolation
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+/*
+ ATTENTION: Rerun code generators when interface signatures are modified.
+*/
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+)
+
+type PodIsolationDetector interface {
+	Detect(vm *v1.VM) (*IsolationResult, error)
+}
+
+type socketBasedIsolationDetector struct {
+	socketDir string
+}
+
+func NewSocketBasedIsolationDetector(socketDir string) PodIsolationDetector {
+	return &socketBasedIsolationDetector{socketDir: socketDir}
+}
+
+func SocketFromNamespaceName(baseDir string, namespace string, name string) string {
+	return filepath.Clean(baseDir) + "/" + namespace + "/" + name + "/sock"
```

Comment:
This is a security risk.

There are eventually few checks on what `name` can contain. An alterantive is to use the VM's UUID

Reply:
Once you have a new way to create the path names, please document it somewhere.

A similar mechanism is actually described here: https://github.com/kubernetes/community/pull/589/files#diff-c1f7e2debf21346978bae50b5f7e37d2R91